packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -1175,98 +1405,765 @@ void main() {
+            // The 48.0 value come from icon min interactive width and height.
+            expect(getInputRect(tester).left, 48.0);
+          });
+
+          testWidgets('Label for outlined decoration aligns horizontally with input when alignLabelWithHint is true', (WidgetTester tester) async {
+            // Regression test for https://github.com/flutter/flutter/issues/113537.
+            await tester.pumpWidget(
+              buildInputDecorator(
+                decoration: const InputDecoration(
+                  prefixIcon: Icon(Icons.ac_unit),
+                  labelText: labelText,
+                  border: OutlineInputBorder(),
+                  alignLabelWithHint: true,
+                ),
+                isFocused: true,
+              ),
+            );
+
+            expect(getLabelRect(tester).left, getInputRect(tester).left);
+          });
+
+          testWidgets('Label for filled decoration is horizontally aligned with text by default', (WidgetTester tester) async {
+            // Regression test for https://github.com/flutter/flutter/issues/113537.
+            // See https://github.com/flutter/flutter/pull/115540.
+            await tester.pumpWidget(
+              buildInputDecorator(
+                decoration: const InputDecoration(
+                  prefixIcon: Icon(Icons.ac_unit),
+                  labelText: labelText,
+                  filled: true,
+                ),
+                isFocused: true,
+              ),
+            );
+
+            // Label and input are horizontally aligned despite `alignLabelWithHint` being false (default value).
+            // The reason is that `alignLabelWithHint` was initially intended for vertical alignement only.
+            // See https://github.com/flutter/flutter/pull/24993 which introduced `alignLabelWithHint` parameter.
+            // See https://github.com/flutter/flutter/pull/115409 which used `alignLabelWithHint` for
+            // horizontal alignment in outlined text field.
+            expect(getLabelRect(tester).left, getInputRect(tester).left);
+          });
+        });
+      });
+    });
+  });
+
+  group('Material3 - InputDecoration helper/counter/error', () {
+    // Overall height for InputDecorator (filled or outlined) is 80dp on mobile:
+    //    8 - top padding
+    //   12 - floating label (font size = 16 * 0.75, line height is forced to 1.0)
+    //    4 - gap between label and input
+    //   24 - input text (font size = 16, line height = 1.5)
+    //    8 - bottom padding
+    //    8 - gap above supporting text
+    //   16 - helper/counter (font size = 12, line height is 1.5)
+    const double topPadding = 8.0;
+    const double floatingLabelHeight = 12.0;
+    const double labelInputGap = 4.0;
+    const double inputHeight = 24.0;
+    const double bottomPadding = 8.0;
+    // TODO(bleroux): make the InputDecorator implementation compliant with M3 spec by changing
+    // the helperGap to 4.0 instead of 8.0.
```

Comment:
Nit: Is there an issue you could link for this?

Reply:
Add a link to https://github.com/flutter/flutter/issues/144984, and currently working on a PR for it.